code/modules/library/skill_learning/skill_station.dm

```diff
@@ -0,0 +1,208 @@
+#define SKILLCHIP_IMPLANT_TIME 1 MINUTES
+#define SKILLCHIP_REMOVAL_TIME 30 SECONDS
+
+/obj/machinery/skill_station
+	name = "Skillsoft Station"
+	desc = "learn skills with only minimal chance for brain damage."
+
+	icon = 'icons/obj/machines/implantchair.dmi'
+	icon_state = "implantchair"
+	occupant_typecache = list(/mob/living/carbon) //todo make occupant_typecache per type
+	state_open = TRUE
+	interaction_flags_atom = INTERACT_ATOM_ATTACK_HAND //Don't call ui_interac by default - we only want that when inside
+	circuit = /obj/item/circuitboard/machine/skill_station
+	/// Currently implanting/removing
+	var/working = FALSE
+	/// Timer until implanting/removing finishes.
+	var/work_timer
+	/// What we're implanting
+	var/obj/item/skillchip/inserted_skillchip
+
+/obj/machinery/skill_station/Initialize()
+	. = ..()
+	update_icon()
+
+//Only usable by the person inside
+/obj/machinery/skill_station/ui_state(mob/user)
+	return GLOB.contained_state
+
+/obj/machinery/skill_station/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "SkillStation", name)
+		ui.set_autoupdate(FALSE)
+		ui.open()
+
+/obj/machinery/skill_station/update_icon_state()
+	icon_state = initial(icon_state)
+	if(state_open)
+		icon_state += "_open"
+	if(occupant)
+		icon_state += "_occupied"
+
+/obj/machinery/skill_station/update_overlays()
+	. = ..()
+	if(working)
+		. += "working"
+
+/obj/machinery/skill_station/relaymove(mob/user as mob)
+	open_machine()
+
+/obj/machinery/skill_station/open_machine()
+	. = ..()
+	interrupt_operation()
+
+/obj/machinery/skill_station/Exited(atom/movable/AM, atom/newloc)
+	. = ..()
+	if(AM == inserted_skillchip)
+		inserted_skillchip = null
+		interrupt_operation()
+
+/obj/machinery/skill_station/power_change()
+	. = ..()
+	if(working)
+		interrupt_operation()
+
+/obj/machinery/skill_station/close_machine(atom/movable/target)
+	. = ..()
+	if(occupant)
+		ui_interact(occupant)
+
+/obj/machinery/skill_station/proc/interrupt_operation()
+	working = FALSE
+	if(work_timer)
+		deltimer(work_timer)
+		work_timer = null
+	update_icon()
+
+/obj/machinery/skill_station/interact(mob/user)
+	. = ..()
+	if(user == occupant)
+		ui_interact(user)
+	else
+		toggle_open()
+
+/obj/machinery/skill_station/attackby(obj/item/I, mob/living/user, params)
+	if(istype(I,/obj/item/skillchip))
+		if(inserted_skillchip)
+			to_chat(user,"<span class='notice'>There's already a skillchip inside.</span>")
+			return
+		else
+			if(!user.transferItemToLoc(I, src))
+				return
+			inserted_skillchip = I
+			SStgui.update_uis(src)
+			return
+	return ..()
+
+/obj/machinery/skill_station/dropContents(list/subset)
+	. = ..(contents - inserted_skillchip) //This is kinda annoying
```

Comment:
```suggestion
/obj/machinery/skill_station/dropContents(list/subset)
	subset = contents - inserted_skillchip
	return ..() //This is kinda annoying
```
This is far from ideal. It's a bit more clear this way, but ideally this behavior should be defined on the object in a more generic way.

Reply:
Was considering creating some common component for these but that was bit out of scope.